player/player.go

```diff
@@ -396,23 +396,33 @@ func (p *Player) PlayPause() error {
 		}
 		return nil
 	case Playing:
-		err := p.setPaused(true)
-		if err == nil {
-			p.prePausedState = p.status.State
-			p.setState(Paused)
-		}
-		return err
+		return p.Pause()
 	case Paused:
-		err := p.setPaused(false)
-		if err == nil {
-			p.setState(p.prePausedState)
-		}
-		return err
+		return p.Continue()
 	default:
 		return errors.New("Unknown player state")
 	}
 }
 
+// Pause playback and update the player state
+func (p *Player) Pause() error {
+	err := p.setPaused(true)
+	if err == nil {
+		p.prePausedState = p.status.State
+		p.setState(Paused)
+	}
+	return err
+}
+
+// Continue playback and update the player state
+func (p *Player) Continue() error {
```

Comment:
Should the behavior of this be different depending on playback state? ie:

````
if Paused {
   do the current impl
} else if Stopped {
   p.PlayFromBeginning()
} // else no-op since already playing
````

Reply:
Here I moved the pre-existing behaviour (previously it lived in the 'PlayPause' switch statement) and exported it in the new function. This is because MacOS has a requirement to set this state explicitly (in addition to being able to toggle).

Happy to change this if we decide the behaviour should work differently!

Reply:
I was referring to the idea that even though this function is just to Play, and not toggle Play/Pause, the behavior should be different based on the current state - if it's called when the player is paused it needs to clear the pause flag (current implementation) but if it's already playing than it's a no-op, and if it's played through the queue and is now stopped, it should restart from the beginning

Reply:
Updated 'Continue()' adding this